File: concrete-sample/src/main/java/com/jaynewstrom/concretesample/details/DetailsListAdapter.java
package com.jaynewstrom.concretesample.details;

import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;

final class DetailsListAdapter extends BaseAdapter {

    DetailsListAdapter() {
    }

    @Override public int getCount() {
        return 50;
    }

    @Override public Object getItem(int position) {
        return position;
    }

    @Override public long getItemId(int position) {
        return position;
    }

    @Override public View getView(int position, View convertView, ViewGroup parent) {
        DetailsListItemView listItemView;
        if (convertView == null) {
            listItemView = new DetailsListItemView(parent.getContext());
        } else {
            listItemView = (DetailsListItemView) convertView;
        }
        listItemView.setPosition(position);
        return listItemView;
    }
}

File: concrete-sample/src/main/java/com/jaynewstrom/concretesample/details/DetailsListAdapter.kt
package com.jaynewstrom.concretesample.details

import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter

internal class DetailsListAdapter : BaseAdapter() {
    override fun getCount(): Int = 50

    override fun getItem(position: Int) = position

    override fun getItemId(position: Int): Long = position.toLong()

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val listItemView: DetailsListItemView = if (convertView == null) {
            DetailsListItemView(parent.context)
        } else {
            convertView as DetailsListItemView
        }
        listItemView.setPosition(position)
        return listItemView
    }
}
